src/gax.ts

```diff
@@ -292,23 +312,147 @@ export class CallSettings {
   }
 }
 
+/**
+ * Validates passed retry options in preparation for eventual parameter deprecation
+ * converts retryRequestOptions to retryOptions
+ * then sets retryRequestOptions to null
+ *
+ * @param {CallOptions} options - a list of passed retry option
+ * @return {CallOptions} A new CallOptions object.
+ *
+ */
+export function checkRetryOptions(
+  options?: CallOptions,
+  gaxStreamingRetries?: boolean
+): CallOptions | undefined {
+  // options will be undefined if no CallOptions object is passed at call time
+  if (!options) {
+    return options;
+  }
+  // if a user provided retry AND retryRequestOptions at call time, throw an error
+  // otherwise, convert supported parameters
+  if (!gaxStreamingRetries) {
+    // if user is opted into legacy settings but has passed retry settings, let them know there might be an issue if it's a streaming call
+    if (options.retry !== undefined) {
+      warn(
+        'legacy_streaming_retry_behavior',
+        'Legacy streaming retry behavior will not honor settings passed at call time or via client configuration. Please set gaxStreamingRetries to true to utilize passed retry settings. gaxStreamingRetries behavior will be set to true by default in future releases.',
+        'DeprecationWarning'
+      );
+    }
+    if (options.retryRequestOptions !== undefined) {
+      warn(
+        'legacy_streaming_retry_request_behavior',
+        'Legacy streaming retry behavior will not honor retryRequestOptions passed at call time. Please set gaxStreamingRetries to true to utilize passed retry settings. gaxStreamingRetries behavior will convert retryRequestOptions to retry parameters by default in future releases.',
+        'DeprecationWarning'
+      );
+    }
+    return options;
+  }
+  if (
+    options.retry !== undefined &&
```

Comment:
I've seen a few places where you check whether something is equal/not equal to undefined. I'm not super familiar with the logic so I'll leave it up to you, but there might be other cases that you miss by not checking truthiness vs. falsiness - i.e., null, [], {}, etc. Just a reminder to check whether you really want to check if it's undefined or if it's falsy!

Reply:
This is a good callout and is something that is definitely tricky and where Alex caught a couple of slipups! I'll do a readthrough of my changes today looking for inadvertent falsiness - in almost all (or maybe even actually all) cases I think I'm looking for definition because most of these are cases where I'm validating presence of a parameter in the config.

Reply:
just did a big runthrough and double check of these. Some do have to stay as "undefined" because they're number parameters that will be `number` or `undefined` and a falsiness check would incorrectly also trip on a zero, but others I was able to make less verbose with a good ol falsiness check. Great callout and a good exercise for me to dig through them.